File: Problem Solving/윤한길/04주차 - 베르트랑 공준.py
import sys
import math

input = sys.stdin.readline


def is_prime(number):
    if number in (0, 1):
        return False
    for i in range(2, int(math.sqrt(number)) + 1):
        if (number % i) == 0:
            return False
    return True


if __name__ == '__main__':
    prime_numbers = [0] * (123456 * 2 + 1)
    for i in range(123456 * 2 + 1):
        if is_prime(i):
            prime_numbers[i] = 1
    while True:
        n = int(input())
        if n == 0:
            break
        print(sum(prime_numbers[n + 1:2 * n + 1]))

File: Problem Solving/김우일/04주차 - 베르트랑 공준.java
package bj;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Bj4948 {

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		while (true) {
			int number = Integer.parseInt(br.readLine());
			if (number == 0)
				break;
			int answer = 0;

			for (int i = number + 1; i <= 2 * number; i++) {
				if (isPrime(i)) {
					answer++;
				}
			}
			System.out.println(answer);
		}
	}

	public static boolean isPrime(int number) {
		if (number == 1)
			return false;
		for (int i = 2; i <= Math.sqrt(number); i++) {
			if (number % i == 0) {
				return false;
			}
		}
		return true;
	}
}
